### ISSUE TITLE
Feature Request - GitHub  actions

### ISSUE THREAD

**Issue by yingzhu146** [pwntester/octo.nvim#149]
## Issue Description

Type: Feature request

### Describe what happened (or what feature you want)

The gh cli now offers a [nice viewer](https://github.blog/2021-04-15-work-with-github-actions-in-your-terminal-with-github-cli/) for actions. As far as I know octo doesn't surface that yet? One of the few workflows I still have to do outside of octo

**Comment by pwntester** [comment]
Thanks for the suggestion, looks like a good addition

**Comment by GustavEikaas** [comment]
Is this on the roadmap? I use octo quite heavily for work and would like this feature. If you are open for it I can make a pull request for it @pwntester? It's quite a big scope but something basic would be a good start

**Comment by wd60622** [comment]
Hi @GustavEikaas, if you can break it down into smaller PRs then that'd be great start and easiest to review. Seems like awesome feature

**Comment by GustavEikaas** [comment]
I tried making something basic just to suit my needs. Could some variant of [gh-actions.nvim](https://github.com/GustavEikaas/gh-actions.nvim) possibly be upstreamed in some shape or form?

**Comment by wd60622** [comment]
That's awesome. Seems really useful. Do you think that it would fit into this repo easily? Maybe starting with `:Octo actions list` command?

What are your thoughts?

**Comment by GustavEikaas** [comment]
Yeah I could take a crack at it, should be pretty straightforward to add I think. By `Octo actions list` do you mean listing out workflow runs or workflow definitions? I think we should try to use the same terminology as github uses. 

From the cli it is 
```
gh run list:          List recent workflow runs
gh workflow list:     List workflow files in your repository
```

**Comment by wd60622** [comment]
Based on my understanding, the separate commands would be easiest to add. Thatd be in the commands.lua file via two new items in table. The alternative would need logic for another level of nesting which I dont think is implemented

**Comment by GustavEikaas** [comment]
Ill have a look at it hopefully today. I agree that having it all under actions would make most sense. Ill look into how much work it is to support this.

**Comment by GustavEikaas** [comment]
So at this point, I have moved the code from my plugin into octo and created the `:Octo run list` command. This is in the draft PR #638 

There are multiple things im planning to do and the code is a mess at this point but please feel free to test out the pr to look for changes/improvements to the UI/DX. I have only ever used this for personal/work projects so there are probably plenty of ux/ui/dx issues. 

To test it
- Start a workflow in a repo
- `:Octo run list`
- `<leader>o` to open and track a specific run
- Also try to open a failing workflow run to see the error message

![image](https://github.com/user-attachments/assets/66346ada-7f89-4323-9586-e545fa4b73e1)



Again the code is a hot mess which I will fix so dont code review it yet 😉
